src/util/fs.js

```diff
@@ -163,6 +163,7 @@ async function buildActionsForCopy(
     const {src, dest, type} = data;
     const onFresh = data.onFresh || noop;
     const onDone = data.onDone || noop;
+    invariant(!files.has(dest), `The same file ${dest} can't be copied twice in one bulk copy`);
```

Comment:
Yeah I'm wrong, I thought it was an array (in which case it would be iterating over the array for each element to copy, hence square complexity), but with a set it's all fine 👍